databricks/koalas/tests/test_groupby.py

```diff
@@ -370,28 +371,34 @@ def test_apply(self):
             kdf.groupby("b").apply(1)
 
     def test_apply_with_new_dataframe(self):
-        # Less than 1000 records will execute a shortcut by using collected pandas dataframe
-        # directly.
-        pdf = pd.DataFrame({
-            "timestamp": [0.0, 0.5, 1.0, 0.0, 0.5],
-            "car_id": ['A', 'A', 'A', 'B', 'B']
-        })
-        kdf = koalas.DataFrame(pdf)
-
-        self.assert_eq(
-            kdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index(),
-            pdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index())
-
-        # 1000+ records will only infer the schema.
-        pdf = pd.DataFrame({
-            "timestamp": [0.0, 0.5, 1.0, 0.0, 0.5],
-            "car_id": ['A', 'A', 'A', 'B', 'B']
-        })
-        kdf = koalas.DataFrame(pdf)
-
-        self.assert_eq(
-            kdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index(),
-            pdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index())
+        # Less than 'compute.shortcut_limit' will execute a shortcut
+        # by using collected pandas dataframe directly.
+        # now we set the 'compute.shortcut_limit' as 1000 explicitly
+        set_option('compute.shortcut_limit', 1000)
+
+        try:
+            pdf = pd.DataFrame({
+                "timestamp": [0.0, 0.5, 1.0, 0.0, 0.5],
+                "car_id": ['A', 'A', 'A', 'B', 'B']
+            })
+            kdf = koalas.DataFrame(pdf)
+
+            self.assert_eq(
+                kdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index(),
+                pdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index())
+
+            # 1000+ records will only infer the schema.
+            pdf = pd.DataFrame({
+                "timestamp": [0.0, 0.5, 1.0, 0.0, 0.5],
+                "car_id": ['A', 'A', 'A', 'B', 'B']
+            })
+            kdf = koalas.DataFrame(pdf)
+
+            self.assert_eq(
+                kdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index(),
+                pdf.groupby('car_id').apply(lambda _: pd.DataFrame({"column": [0.0]})).sort_index())
+        finally:
+            reset_option('compute.shortcut_limit')
 
     def test_transform(self):
```

Comment:
And, while we're here, can you remove those tests in `test_filter`:

```
        # Data is intentionally big to test when schema inference is on.
        pdf = pd.DataFrame({'a': [1, 2, 3, 4, 5, 6] * 300,
                            'b': [1, 1, 2, 3, 5, 8] * 300,
                            'c': [1, 4, 9, 16, 25, 36] * 300}, columns=['a', 'b', 'c'])
        kdf = koalas.DataFrame(pdf)
        self.assert_eq(kdf.groupby("b").filter(lambda x: x.b.mean() < 4).sort_index(),
                       pdf.groupby("b").filter(lambda x: x.b.mean() < 4).sort_index())
        self.assert_eq(kdf.groupby(['a', 'b']).filter(lambda x: any(x.a == 2)).sort_index(),
                       pdf.groupby(['a', 'b']).filter(lambda x: any(x.a == 2)).sort_index())
        with self.assertRaisesRegex(TypeError, "<class 'int'> object is not callable"):
            kdf.groupby("b").filter(1)
```

`groupby.filter` actually doesn't infer its schema. It seems mistakenly the test was added.